pkgs/top-level/cuda-packages.nix

```diff
@@ -37,41 +38,35 @@ let
     strings
     versions
     ;
+
   # MUST be defined outside fix-point (cf. "NAMESET STRICTNESS" above)
   fixups = import ../development/cuda-modules/fixups { inherit lib; };
-  gpus = import ../development/cuda-modules/gpus.nix;
-  nvccCompatibilities = import ../development/cuda-modules/nvcc-compatibilities.nix;
-  flags = import ../development/cuda-modules/flags.nix {
-    inherit
-      config
-      cudaMajorMinorVersion
-      gpus
-      lib
-      stdenv
-      ;
-  };
 
-  mkVersionedPackageName =
-    name: version: name + "_" + strings.replaceStrings [ "." ] [ "_" ] (versions.majorMinor version);
+  # Since Jetson capabilities are never built by default, we can check if any of them were requested
+  # through final.config.cudaCapabilities and use that to determine if we should change some manifest versions.
+  # Copied from backendStdenv.
+  jetsonCudaCapabilities = lib.filter (
+    cudaCapability: cudaLib.data.cudaCapabilityToInfo.${cudaCapability}.isJetson
+  ) cudaLib.data.allSortedCudaCapabilities;
+  hasJetsonCudaCapability =
+    lib.intersectLists jetsonCudaCapabilities (config.cudaCapabilities or [ ]) != [ ];
+  redistSystem = cudaLib.utils.getRedistSystem hasJetsonCudaCapability stdenv.hostPlatform.system;
 
   passthruFunction = final: {
     inherit
+      cudaLib
```

Comment:
Spotted! I saw some occurrences of `final.cudalib`, but I it was inside the library, referring to the `makeExtensible` thing.

This bit, I believe, is very important: `cudaPackages` should not have their own `cudaLib` fixpoints because then they can have different (diverging) "libraries"; in the current version of the PR, `callPackage` should be picking `cudaLib` from the top-level! I think it's important that the "library" (esp. the data) be not affected by the "configuration" of any particular `cudaPackages` set. The layering should be: `configuration` `->` (identifies/is part of) `cudaPackages` `->` (is built using) `cudaLib`. No arrows should point in the reverse directions

Reply:
So all `cudaLib` knows about "package sets" and "configurations" is it implements functions for interpreting configurations and resolving them against data

Reply:
I like how you phrased that -- in the latest I added a comment in `pkgs/top-level/cuda-packages.nix` which says about as much.